File: app/src/main/java/com/shevelev/comics_viewer/activities/folders/file_system/folders_tree/IFoldersTreeItemsGetter.java
package com.shevelev.comics_viewer.activities.folders.file_system.folders_tree;

import com.shevelev.comics_viewer.common.threads.ICancelationTokenRead;

import java.util.List;

/**
 * Interface of class for getting tree items
 */
public interface IFoldersTreeItemsGetter
{
    List<FoldersTreeItem> getSubItems(ICancelationTokenRead cancelationToken);
}

File: app/src/main/java/com/shevelev/comics_viewer/activities/folders/file_system/folders_tree/IFoldersTreeItemsGetter.kt
package com.shevelev.comics_viewer.activities.folders.file_system.folders_tree

import com.shevelev.comics_viewer.core.cancelation.ICancelationTokenRead

/**
 * Interface of class for getting tree items
 */
interface IFoldersTreeItemsGetter {
    fun getSubItems(cancellationToken: com.shevelev.comics_viewer.core.cancelation.ICancelationTokenRead): List<FoldersTreeItem>?
}